dvc/repo/artifacts.py

```diff
@@ -1,43 +1,57 @@
 import logging
-import os
 import re
-from typing import TYPE_CHECKING, Dict
+from pathlib import Path
+from typing import Dict, Optional
 
 from dvc.annotations import Artifact
-from dvc.dvcfile import FileMixin
+from dvc.dvcfile import PROJECT_FILE
+from dvc.exceptions import InvalidArgumentError
+from dvc.repo import Repo
 from dvc.utils import relpath
-
-if TYPE_CHECKING:
-    from dvc.repo import Repo
+from dvc.utils.serialize import modify_yaml
 
 logger = logging.getLogger(__name__)
 
 
-NAME_RE = re.compile(r"^[a-z]([a-z0-9-]*[a-z0-9])?$")
+# Constants are taken from GTO.
+# When we make it a dependency, we can import them instead
+SEPARATOR_IN_NAME = ":"
+DIRNAME = r"[a-z0-9-_./]+"
+NAME = r"[a-z0-9]([a-z0-9-/]*[a-z0-9])?"
+NAME_RE = re.compile(f"^{NAME}$")
+FULLNAME = f"((?P<dirname>{DIRNAME}){SEPARATOR_IN_NAME})?(?P<name>{NAME})"
+FULLNAME_RE = re.compile(f"^{FULLNAME}$")
 
 
 def name_is_compatible(name: str) -> bool:
     return bool(NAME_RE.search(name))
 
 
+wrong_name_message = (
+    "Can't use '%s' as artifact name (ID)."
```

Comment:
Use format strings where possible.
```suggestion
    "Can't use {0!r} as artifact name (ID)."
```
And below, use `wrong_name_message.format(name)